src/blockdev.rs

```diff
@@ -14,35 +14,57 @@
 
 use error_chain::bail;
 use nix::errno::Errno;
-use nix::{self, ioctl_none, ioctl_write_ptr_bad, mount, request_code_none};
+use nix::{self, ioctl_none, ioctl_read_bad, ioctl_write_ptr_bad, mount, request_code_none};
 use regex::Regex;
 use std::collections::HashMap;
+use std::convert::TryInto;
 use std::fs::{remove_dir, File};
-use std::io::{Seek, SeekFrom};
+use std::io::{self, BufRead, BufReader, Read, Seek, SeekFrom};
+use std::os::raw::c_int;
 use std::os::unix::io::AsRawFd;
 use std::path::{Path, PathBuf};
 use std::process::Command;
+use std::result;
 use std::thread::sleep;
 use std::time::Duration;
 use tempdir::TempDir;
 
 use crate::errors::*;
 
 pub fn mount_boot(device: &str) -> Result<Mount> {
-    let dev = get_partition_with_label(device, "boot")?
-        .chain_err(|| format!("couldn't find boot device for {}", device))?;
-    match dev.fstype {
+    // get partition list
+    let partitions = get_partitions(device)?;
+    if partitions.is_empty() {
+        bail!("couldn't find any partitions on {}", device);
+    }
+
+    // find the boot partition
+    let boot_partitions = partitions
+        .iter()
+        .filter(|d| d.label.as_ref().unwrap_or(&"".to_string()) == "boot")
+        .collect::<Vec<&BlkDev>>();
+    if boot_partitions.is_empty() {
+        bail!("couldn't find boot device for {}", device);
+    } else if boot_partitions.len() > 1 {
```

Comment:
Self-note: `itertools` has a few helper methods to collapse a collection to a single-matching-item, but none of them with ergonomic empty-vs-many logic (and we don't currently bring in that crate, not even transitively).